enterprise/internal/codeintel/gitserver/client.go

```
@@ -75,19 +75,17 @@ func (c *Client) CommitsExist(ctx context.Context, commits []RepositoryCommit) (
 		return nil, err
 	}
 
-	exists := make([]bool, len(commits))
-	for i, rc := range commits {
-		name, ok := repositoryNames[rc.RepositoryID]
-		if !ok {
-			continue
-		}
-
-		e, err := git.CommitExists(ctx, c.db, name, api.CommitID(rc.Commit), authz.DefaultSubRepoPermsChecker)
-		if err != nil {
-			return nil, err
-		}
+	repoCommits := make([]api.RepoCommit, 0, len(commits))
+	for _, rc := range commits {
+		repoCommits = append(repoCommits, api.RepoCommit{
+			Repo:     repositoryNames[rc.RepositoryID],
```

Q: Do you think it is worth guarding against an invariant failure here where the `DBStore`'s `RepoNames` method is incorrectly implemented, returning a map which doesn't cover all the repository IDs? If so, perhaps we should return an error here if the map access fails.

---

Non-blocking suggestion: For this point specifically, I was thinking of something like `if repo, ok := repositoryNames[rc.RepositoryID]; !ok { /* return error */ }`, but yeah, handling it at the lower layer (probably at the right place) would require some more changes.

---

Oh that's probably a decent solution if we need to get this in before doing it in a way that keeps partial results in mind. I think either way has an error will blow up this particular method call, so we won't lose anything with an interim solution.

Whatever I end up doing for this PR I'll make sure we have a follow-up issue if necessary.

---

@varungandhi-src Can you take a look at ae8e6e506f89e374c8b570e86efa3dc0fc6ec0fb?